src/proxy_handler.cc

```diff
@@ -0,0 +1,160 @@
+
+		    if(findLoc != -1){
+		    	uri.erase(findLoc, uri_prefix.size());
+		    }
+
+		    std::cout << "URIBEINGSENT:  " << uri << std::endl;
+ 		    boost::asio::streambuf requestBuf;
+		    std::ostream request_stream(&requestBuf);
+		    request_stream << "GET " << uri << " HTTP/1.1\r\n";
+		    request_stream << "Host: " << host_name << ":" << portno << "\r\n"; 
+		    request_stream << "Accept: */*\r\n";
+		    request_stream << "Connection: close\r\n\r\n";
+
+		    // Send the request.
+		    boost::asio::write(socket, requestBuf);
+
+		    // Read the response status line. The response streambuf will automatically
+		    // grow to accommodate the entire line. The growth may be limited by passing
+		    // a maximum size to the streambuf constructor.
+		    boost::asio::streambuf resp;
+		    boost::asio::read_until(socket, resp, "\r\n");
+
+		    // Check that response is OK.
+		    std::istream response_stream(&resp);
+		    std::string http_version;
+		    response_stream >> http_version;
+		    std::cout << "http version: " << http_version << std::endl;
+
+		    unsigned int status_code;
+		    response_stream >> status_code;
+		    std::cout << "status code: " << status_code << std::endl;
+
+		    std::string status_message;
+		    std::getline(response_stream, status_message);
+		    std::cout << "status message: " << status_message << std::endl;
+		    
+		    if (!response_stream || http_version.substr(0, 5) != "HTTP/")
+		    {
+		      std::cout << "Invalid response\n";
+		      //return 1;
+		    }
+		    if (status_code == 200)
+		    {
+		      std::cout << "Response returned with status code " << status_code << "\n";
+		      response->SetStatus(Response::ok);
+		      //return 1;
+		    }
+		    else if(status_code == 302 || status_code == 301){
+		      response->SetStatus(Response::redirect);
+		      //return RequestHandler::handle_fail;
+		    }
+		    else if(status_code == 404){
+		      response->SetStatus(Response::not_found);
+		      return RequestHandler::handle_fail;
+		    }
+		    else {
+		      response->SetStatus(Response::bad_request);
+		      return RequestHandler::handle_fail;
+		    }
+
+
+		    // Read the response headers, which are terminated by a blank line.
+		    boost::asio::read_until(socket, resp, "\r\n\r\n");
+
+		    // Process the response headers.
+		    std::string header;
+		    while (std::getline(response_stream, header) && header != "\r"){
+		        std::cout << "response header: " << header << "\n";
+				std::string head = header.substr(0, header.find(":"));
+				std::string value = header.substr(header.find(":")+2);
+				if(head == "Location"){
+					int loc = value.find_last_of("/");
+					value = value.substr(0, loc+1);
+					std::cout << value << std::endl;
+				}
+				std::cout << head << "," << value << std::endl;
+				response->AddHeader(head, value);
+		    }
+
+		    // Read until EOF, writing data to output as we go.
+		    boost::system::error_code error;
```

Comment:
if you do this way(read until eof), maybe you will meet problems like piazza said ( "image files are allowed to have EOF characters"). Instead I recommend you guys read the "Content-Length" from the header, and use read(s, buffer, transfer_exactly(...), ec)